pkg/loghttp/push/push.go

```diff
@@ -206,12 +234,29 @@ func ParseLokiRequest(userID string, r *http.Request, tenantsRetention TenantsRe
 
 	for _, s := range req.Streams {
 		pushStats.streamLabelsSize += int64(len(s.Labels))
-		var retentionPeriod time.Duration
-		if tenantsRetention != nil {
-			lbs, err := syntax.ParseLabels(s.Labels)
+
+		// Init custom trackers
+		customTrackers := limits.CustomTrackersConfig(userID)
+		var lbs labels.Labels
+		if tenantsRetention != nil || len(customTrackers.source) > 0 {
+			lbs, err = syntax.ParseLabels(s.Labels)
 			if err != nil {
 				return nil, nil, fmt.Errorf("couldn't parse labels: %w", err)
 			}
+		}
+		trackedLabels := customTrackers.MatchTrackers(lbs)
```

Comment:
I would call it trackers
```suggestion
		trackers := customTrackers.MatchTrackers(lbs)
```